### ISSUE TITLE
Wrong version detection in Reader-broken-databases.t

### ISSUE THREAD

**Issue by madogiwa** [maxmind/MaxMind-DB-Reader-XS#21]
I encountered the following error during ```$ cpanm MaxMind::DB::Reader::XS```.

```
t/00-report-prereqs.t ...................... ok

#   Failed test 'expected exception with unknown file type'
#   at t/MaxMind/DB/Reader-broken-databases.t line 84.
#                   'MaxMind::DB::Reader::XS - Error opening database file "/tmp/9AJiT3aWv9/garbage": The MaxMind DB file contains invalid metadata at /root/.cpanm/work/1519046793.22989/MaxMind-DB-Reader-XS-1.000004/blib/lib/MaxMind/DB/Reader/XS.pm line 67.
# '
#     doesn't match '(?^:Error opening database file "\/tmp\/9AJiT3aWv9\/garbage": The lookup path does not match the data .+)'
# Looks like you failed 1 test of 5.
t/MaxMind/DB/Reader-broken-databases.t .....
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/5 subtests
t/MaxMind/DB/Reader-decoder.t .............. ok
t/MaxMind/DB/Reader-no-ipv4-search-tree.t .. ok
t/MaxMind/DB/Reader.t ...................... ok
t/MaxMind/DB/Reader/NoMoose.t .............. ok
# libmaxminddb version is 1.3.2
t/libmaxminddb-version.t ................... ok
```

My libmaxminddb version is 1.3.2 (latest release version). 

Reader-broken-databases.t uses the patch level to distinguish whether it is an older version or not.
Version 1.3.2 is probably mistaken as an older version.

https://github.com/maxmind/MaxMind-DB-Reader-XS/blob/527804fd925d6853a2446917905629a814c395ab/t/MaxMind/DB/Reader-broken-databases.t#L70-L77

**Comment by oschwald** [comment]
I am not seeing this on my machine with 1.3.2. What do you get when you run the following?

```
perl -MMaxMind::DB::Reader::XS -le 'print MaxMind::DB::Reader::XS::libmaxminddb_version()'
```

**Comment by madogiwa** [comment]
The execution result is follows.

```
# perl -MMaxMind::DB::Reader::XS -le 'print MaxMind::DB::Reader::XS::libmaxminddb_version()'
Can't locate MaxMind/DB/Reader/XS.pm in @INC (@INC contains: /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .).
BEGIN failed--compilation aborted.
```

So, I installed MaxMind::DB::Reader::XS by skipping tests.

```
# cpanm --notest MaxMind::DB::Reader::XS
--> Working on MaxMind::DB::Reader::XS
Fetching http://www.cpan.org/authors/id/M/MA/MAXMIND/MaxMind-DB-Reader-XS-1.000004.tar.gz ... OK
Configuring MaxMind-DB-Reader-XS-1.000004 ... OK
Building MaxMind-DB-Reader-XS-1.000004 ... OK
Successfully installed MaxMind-DB-Reader-XS-1.000004
1 distribution installed
```

The result of executing the command again is follows.

```
# perl -MMaxMind::DB::Reader::XS -le 'print MaxMind::DB::Reader::XS::libmaxminddb_version()'
1.3.2
```

**Comment by oschwald** [comment]
I have been unable to reproduce this locally. Reading the code, it isn't obvious to me why it might be failing for you.

**Comment by madogiwa** [comment]
I have been able to reproduce my problem from a clean install.
How to reproduce using Docker is as follows.

1. Run Docker with latest CentOS official image.
2. Install cpanm and gcc.
3. Install libmaxminddb from https://github.com/maxmind/libmaxminddb/releases
4. cpanm install MaxMind::DB::Reader::XS

Example commands are as following.

```
$ docker run -i -t centos /bin/bash
[root@3297fa2edceb /]# yum install -y gcc perl-App-cpanminus
[root@3297fa2edceb /]# curl -L https://github.com/maxmind/libmaxminddb/releases/download/1.3.2/libmaxminddb-1.3.2.tar.gz | tar xz
[root@3297fa2edceb /]# cd libmaxminddb-1.3.2 && ./configure --prefix=/usr && make && make install && ldconfig
[root@3297fa2edceb libmaxminddb-1.3.2]# cpanm MaxMind::DB::Reader::XS

(snip)

Building and testing MaxMind-DB-Reader-1.000013 ... OK
Successfully installed MaxMind-DB-Reader-1.000013
Building and testing MaxMind-DB-Reader-XS-1.000004 ... FAIL
! Installing MaxMind::DB::Reader::XS failed. See /root/.cpanm/work/1519451239.2982/build.log for details. Retry with --force to force install it.
112 distributions installed
```

The same error recorded in build.log.

```
[root@3297fa2edceb libmaxminddb-1.3.2]# cat /root/.cpanm/work/1519451239.2982/build.log  | grep -A 8 "Failed test 'expected exception with unknown file type'"
#   Failed test 'expected exception with unknown file type'
#   at t/MaxMind/DB/Reader-broken-databases.t line 84.
#                   'MaxMind::DB::Reader::XS - Error opening database file "/tmp/tdnlaV_qus/garbage": The MaxMind DB file contains invalid metadata at /root/.cpanm/work/1519451239.2982/MaxMind-DB-Reader-XS-1.000004/blib/lib/MaxMind/DB/Reader/XS.pm line 67.
# '
#     doesn't match '(?^:Error opening database file "\/tmp\/tdnlaV_qus\/garbage": The lookup path does not match the data .+)'
# Looks like you failed 1 test of 5.
t/MaxMind/DB/Reader-broken-databases.t .....
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/5 subtests
```

**Comment by oschwald** [comment]
Closed via #22. Thanks!
